### PR TITLE
New WebXR API using surfman

### LINKED ISSUE #60: Use surfman for surface sharing
Surfman is the next-generation of offscreen-gl-context, with support for surface sharing. We should use it rather than texture sharing and `GLSync` in webxr.

### PR DESCRIPTION
**Pull request by asajeffrey**
This is a surfman-aware webxr. Fixes #60

### CODE REVIEW

**Comment by asajeffrey** [comment]
Probably shouldn't be merged until servo+surfman is ready to go.

**Comment by asajeffrey** [comment]
Note: I've only tested the glwindow implementation!

**Comment by asajeffrey** [comment]
That would be the drop code for a surface being run without destroy'ing it first.

**Comment by jdm** [comment]
OpenXrDevice contains an Option<(Device, Context)>, so I suppose it needs to be more careful about destroying the context before that value is dropped.

**Comment by asajeffrey** [comment]
Ah, that's a pretty easy fix, we can destroy the context in the drop code.

**Comment by asajeffrey** [comment]
@jdm: do you want to try again and see if that fixed it?

**Comment by asajeffrey** [comment]
The matching servo PR is https://github.com/servo/servo/pull/24482

File: webxr-api/registry.rs [hunk 1]
```diff
@@ -185,10 +187,10 @@ impl MainThreadRegistry {
     }
 
     fn request_session(&mut self, mode: SessionMode) -> Result<Session, Error> {
-        let webgl = self.webgl.as_ref().ok_or(Error::NoMatchingDevice)?;
+        let swap_chains = self.swap_chains.as_mut().ok_or(Error::NoMatchingDevice)?;
```
Comment (Manishearth) [inline webxr-api/registry.rs]: maybe a new error variant?

File: webxr/glwindow/mod.rs [hunk 2]
```diff
@@ -202,15 +207,29 @@ impl Device for GlWindowDevice {
...
+    fn drop(&mut self) {
+        let _ = self.device.destroy_context(&mut self.context);
+    }
+}
+
 impl GlWindowDevice {
     fn new(gl: Rc<dyn Gl>, window: Rc<dyn GlWindow>) -> Result<GlWindowDevice, Error> {
         window.make_current();
+        #[cfg(target_os = "windows")]
+        let (device, context) =
```
Comment (Manishearth) [inline webxr/glwindow/mod.rs]: why is this cfg necessary?

Comment (Manishearth) [reply webxr/glwindow/mod.rs]: (leave a comment in the code)

File: webxr-api/registry.rs (hunk 1 above)
Comment (asajeffrey) [reply webxr-api/registry.rs]: Well, I think the spec says the only error this can return is NoMatchingDevice?

Comment (Manishearth) [reply webxr-api/registry.rs]: Sure, but we can be more specific in our API if we want. r=me either way.

File: webxr/glwindow/mod.rs (hunk 2 above)
Comment (asajeffrey) [reply webxr/glwindow/mod.rs]: Done.

**Pull request merged**